src/python/pants/backend/python/rules/pex_from_targets_test.py

```diff
@@ -97,11 +97,11 @@ def get_pex_request(
             )
 
         pex_req1 = get_pex_request("constraints1.txt", ResolveAllConstraintsOption.NEVER)
-        assert pex_req1.requirements == PexRequirements(["foo>=0.1.2", "bar==5.5.5", "baz"])
+        assert pex_req1.requirements == PexRequirements(["foo-bar>=0.1.2", "bar==5.5.5", "baz"])
 
         pex_req2 = get_pex_request("constraints1.txt", ResolveAllConstraintsOption.ALWAYS)
         assert pex_req2.requirements == PexRequirements(
-            ["foo==1.0.0", "bar==5.5.5", "baz==2.2.2", "qux==3.4.5"]
+            ["Foo._-BAR==1.0.0", "bar==5.5.5", "baz==2.2.2", "qux==3.4.5"]
```

Comment:
This test fails without the fix since, without the fix, `Foo._-BAR` is not canonicalized to match `foo-bar` and so `PexRequirements(["foo-bar>=0.1.2", "bar==5.5.5", "baz"])` is fallen back to.